From b0b0ebc01bb005c9cbddd488b5bfadae753a8632 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sun, 19 Aug 2018 08:53:49 -0600
Subject: [PATCH 3/3] DO NOT MERGE: klte-common: Requisite bring-up BS change

Currently:
* Permissive
* Don't build our sepolicy
* Kill contexts in our fstab
* Kill seclabel on macloader service
* Don't build SamsungDoze
* Don't build FlipFlap

Change-Id: Iec4e6927e019c9ec91cc16d3eeacb0990ffff6f5
---
 BoardConfigCommon.mk     | 1 +
 klte.mk                  | 4 ++--
 rootdir/etc/fstab.qcom   | 4 ++--
 rootdir/etc/init.qcom.rc | 1 -
 sepolicy/sepolicy.mk     | 2 +-
 5 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 68355a4..ad54467 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -45,6 +45,7 @@ DEVICE_MANIFEST_FILE += $(COMMON_PATH)/manifest.xml
 # Kernel
 BOARD_KERNEL_BASE := 0x00000000
 BOARD_KERNEL_CMDLINE := console=null androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x37 ehci-hcd.park=3 zcache.enabled=1 zcache.compressor=lz4 maxcpus=1
+BOARD_KERNEL_CMDLINE += androidboot.selinux=permissive
 BOARD_KERNEL_IMAGE_NAME := zImage
 BOARD_KERNEL_PAGESIZE := 2048
 BOARD_KERNEL_SEPARATED_DT := true
diff --git a/klte.mk b/klte.mk
index 55b4dc8..1236cb1 100644
--- a/klte.mk
+++ b/klte.mk
@@ -46,11 +46,11 @@ PRODUCT_AAPT_CONFIG := normal
 PRODUCT_AAPT_PREF_CONFIG := xxhdpi
 
 # Doze
-PRODUCT_PACKAGES += \
+#PRODUCT_PACKAGES += \
     SamsungDoze
 
 # FlipFlap
-PRODUCT_PACKAGES += \
+#PRODUCT_PACKAGES += \
     FlipFlap
 
 # IPv6 tethering
diff --git a/rootdir/etc/fstab.qcom b/rootdir/etc/fstab.qcom
index d45e281..13a890c 100644
--- a/rootdir/etc/fstab.qcom
+++ b/rootdir/etc/fstab.qcom
@@ -11,8 +11,8 @@
 /dev/block/platform/msm_sdcc.1/by-name/userdata     /data           ext4    nosuid,nodev,noatime,noauto_da_alloc,errors=continue,commit=20                                  wait,check,formattable,encryptable=footer,length=-16384
 /dev/block/platform/msm_sdcc.1/by-name/cache        /cache          f2fs    nosuid,nodev,noatime,rw,inline_xattr                                                            wait,check,formattable
 /dev/block/platform/msm_sdcc.1/by-name/cache        /cache          ext4    nosuid,nodev,noatime,noauto_da_alloc,errors=continue,commit=20                                  wait,check,formattable
-/dev/block/platform/msm_sdcc.1/by-name/apnhlos      /firmware       sdfat   ro,uid=1000,gid=1026,dmask=227,fmask=337,context=u:object_r:firmware_file:s0                    wait
-/dev/block/platform/msm_sdcc.1/by-name/modem        /firmware-modem sdfat   ro,uid=1000,gid=1026,dmask=227,fmask=337,context=u:object_r:firmware_file:s0                    wait
+/dev/block/platform/msm_sdcc.1/by-name/apnhlos      /firmware       sdfat   ro,uid=1000,gid=1026,dmask=227,fmask=337                    wait
+/dev/block/platform/msm_sdcc.1/by-name/modem        /firmware-modem sdfat   ro,uid=1000,gid=1026,dmask=227,fmask=337                    wait
 /dev/block/platform/msm_sdcc.1/by-name/efs          /efs            ext4    nosuid,nodev,noatime,noauto_da_alloc,journal_async_commit,errors=panic                          wait,check
 /dev/block/platform/msm_sdcc.1/by-name/fota         /misc           emmc    defaults                                                                                        defaults
 
diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index 8a6a20b..636c179 100644
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -930,7 +930,6 @@ service macloader /vendor/bin/macloader
     oneshot
     user system
     group system wifi
-    seclabel u:r:macloader:s0
 
 on property:wlan.driver.status=ok
     start macloader
diff --git a/sepolicy/sepolicy.mk b/sepolicy/sepolicy.mk
index 36d7b55..8e9102f 100644
--- a/sepolicy/sepolicy.mk
+++ b/sepolicy/sepolicy.mk
@@ -15,5 +15,5 @@
 #
 
 # Board specific SELinux policy variable definitions
-BOARD_SEPOLICY_DIRS += \
+#BOARD_SEPOLICY_DIRS += \
     device/samsung/klte-common/sepolicy/common
-- 
2.17.1

